### PR TITLE
initial import of the CFT Fabric COS modules

### PR DESCRIPTION
**Pull request by ludoo**
Morgante, here is the initial import of the Fabric COS modules, as submodules.

### CODE REVIEW

**Comment by ludoo** [comment]
This is ready for review.

File: modules/cos-coredns/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  corefile       = "assets/Corefile.cloud"
+  vm_tags        = ["ssh"]
+  client_cidrs   = ["0.0.0.0/0"]
+  network        = "my-network-name"
+  subnetwork     = "https://www.googleapis.com/compute/v1/projects/my-project/regions/europe-west3/subnetworks/my-subnetwork"
+}
+```
+
+## Inputs
+
```
Comment (morgante) [inline modules/cos-coredns/README.md]: Should add tags to allow autogenerating this (see main README).

File: modules/cos-coredns/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+* `google_compute_instance`, one or more COS instances
+* `google_compute_firewall`, one firewall rule to allow traffic from specific IP ranges to the DNS port
+
+Variables allow controlling several aspects of the created resurces, like number of instances (`instance_count`), container image used for CoreDNS (`container_image`), client IP ranges allowed to connect to the service(`client_cidrs`), and CoreDNS configuration file (`corefile`).
+
+## Sample Usage
+
+```hcl
+module "dns-service" {
+  source         = "modules/cos-coredns"
```
Comment (morgante) [inline modules/cos-coredns/README.md]: This source won't actually work. Source should be:

```suggestion
  source         = "terraform-google-modules/container-vm/google//modules/cos-coredns"
```

File: modules/cos-generic/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+Variables allow controlling several aspects of the created resurces, like number of instances (`instance_count`), the container image used for CoreDNS (`container_image`), the client IP ranges (`client_cidrs`), the cloud config file passed to the instance (`cloud-init`).
+
+## Sample Usage
+
+This is a sample module invocation:
+
+```hcl
+module "simple-gcs-copier" {
+  source                = "modules/cos-generic"
```
Comment (morgante) [inline modules/cos-generic/README.md]: This source won't actually work. Source should be:

```suggestion
  source         = "terraform-google-modules/container-vm/google//modules/cos-generic"
```

File: modules/cos-mysql/README.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+KMS decryption on the instance leverages the instance service account, which needs the `roles/cloudkms.cryptoKeyDecrypter` role on the provided key. The corresponding KMS scope is added by this module to the instance scopes if a valid KMS configuration is present.
+
+The [official Docker MySQL image](https://hub.docker.com/_/mysql) supports various customization options which are not directly implemented here, but are fairly trivial to add by modifying the `cloud-init` file set on the instance.
+
+## Sample Usage
+
+```hcl
+module "mysql" {
+  source         = "modules/cos-mysql"
```
Comment (morgante) [inline modules/cos-mysql/README.md]: This source won't actually work. Source should be:

```suggestion
  source         = "terraform-google-modules/container-vm/google//modules/cos-mysql"
```

**Comment by morgante** [comment]
@ludoo This looks fine for initial import. Please file a follow-up PR w/ noted changes.

**Pull request merged**